### DISCUSSION TITLE
Create IAM Role With External ID. Its possible?

### DISCUSSION THREAD

**Discussion by evairmarinho** [aws/aws-cli#6951] [Q&A] [answered]
I cannot find this option in documentation

**Comment by tim-finnigan** [comment]
Hi @evairmarinho I found a couple of  documentation pages that might help: 
* https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-user.html
* https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-user_externalid.html

**Reply by evairmarinho** [reply]
Thanks, @tim-finnigan! The problem is that this configuration seems to be possible only via AWS console. I would like to do this creation of the role programmatically using the AWS CLI, or SDK. However, I did not find this external id parameter.

**Reply by tim-finnigan** [reply, marked as answer]
Have you tried using the [create-role](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/iam/create-role.html) command to do this? Based on that documentation it looks like you can define `sts:ExternalId` in the policy you use.
